src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs

```diff
@@ -7,12 +7,10 @@ namespace System.Runtime.InteropServices
 {
     public static partial class RuntimeInformation
     {
-        private static readonly object s_osLock = new object();
-        private static readonly object s_processLock = new object();
         private static string? s_osPlatformName;
         private static string? s_osDescription;
-        private static Architecture? s_osArch;
-        private static Architecture? s_processArch;
+        private static readonly Architecture s_osArch = Map((Interop.Sys.ProcessorArchitecture)Interop.Sys.GetOSArchitecture());
+        private static readonly Architecture s_processArch = Map((Interop.Sys.ProcessorArchitecture)Interop.Sys.GetProcessArchitecture());
```

Comment:
These two P/Invokes also seem like perfect candidates for SuppressGCTransitionAttribute.  It might be worthwhile doing an audit of our shims looking for places it can easily be applied.

Reply:
cc: @AaronRobinsonMSFT

Reply:
Oh wow. Those functions are exactly what `SuppressGCTransitionAttribute` was designed to improve. Just put the `[SuppressGCTransition]` on the P/Invoke.

Reply:
done